### PR TITLE
FIXED -> Support bold and italicized text in CopilotTextarea #62

### PR DESCRIPTION
**Pull request by zsskar**
add functionality as per the requirement please check it if you find it working as per the requirement please merge the code.

### CODE REVIEW

File: CopilotKit/packages/react-textarea/src/types/base/custom-editor.tsx [hunk 1]
```diff
@@ -16,7 +16,17 @@ export type SuggestionElement = {
   children: CustomText[];
 };
 
-export type CustomElement = ParagraphElement | SuggestionElement;
+export type BoldElement = {
```
Comment (ataibarkai) [inline CopilotKit/packages/react-textarea/src/types/base/custom-editor.tsx:19]: looks good

File: CopilotKit/packages/react-textarea/src/components/base-copilot-textarea/base-copilot-textarea.tsx [hunk 2]
```diff
@@ -128,8 +128,54 @@ const BaseCopilotTextareaWithHoveringContext = React.forwardRef(
...
+    >
+      {props.children}
+    </em>
+  );
+};
+
+
     const renderElementMemoized = useMemo(() => {
-      return makeRenderElementFunction(suggestionStyleAugmented);
+      return makeRenderElementFunction({
```
Comment (ataibarkai) [inline CopilotKit/packages/react-textarea/src/components/base-copilot-textarea/base-copilot-textarea.tsx]: looks like `makeRenderElementFunction` still only takes a single input (`suggestionStyle`).
So passing it `boldElement` and `italicElement` is incorrect.

We have 2 options here:
1. The pre-existing approach uses CSS styling. We can mirror what we have for the suggestion element with bold/italics 

2. I actually like the approach you started here - using <strong> and <em> to demarcate bold/italics, and carrying over the remaining stylings.

Assuming it works, there's no need to pass boldElement / italicsElement to the `makeRenderElementFunction` function.

**Review by ataibarkai — COMMENTED** [review]
Thank you. @zsskar !

Looks like it's heading in a good direction.

A couple of related notes:
1. looks like there are some build errors. I added a section to the [contributing docs quick-start](https://docs.copilotkit.ai/contribute/quickstart-contribute) with how to test your solution locally

2. Let's get a running example so we can see it working in action.
You can modify the existing /textarea example, to get text editing working.

We also need to create UI hooks for enabling/disabling bolding, but we can address that separately. For the purposes of the running example, it can be hard-coded.

Let me know if you need additional support on this!

File: CopilotKit/packages/react-textarea/src/components/base-copilot-textarea/base-copilot-textarea.tsx (hunk 2 above)
Comment (zsskar) [reply CopilotKit/packages/react-textarea/src/components/base-copilot-textarea/base-copilot-textarea.tsx]: Ya sure ataibarkai I will first setup it locally and test it on my local and try to make changes what you have mentioned. If I need any help definitely I will ask you.

Thanks.

**Pull request closed without merging**